sdk/storage/azure-storage-blob/src/main/java/com/azure/android/storage/blob/transfer/StorageBlobClientsMap.java

```diff
@@ -0,0 +1,55 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.android.storage.blob.transfer;
+
+import com.azure.android.storage.blob.StorageBlobClient;
+
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+/**
+ * Package private.
+ *
+ * A map of configured {@link StorageBlobClient} used by {@link UploadHandler} instances to
+ * make API calls.
+ */
+final class StorageBlobClientsMap {
```

Comment:
As discussed in our arch chat, we'll want to think about this. If we're coming back from the app being closed, the client map would be empty (since we can't persist it to the database) and so the resume would (presumably) fail.

We probably need to permit multiple clients to be "registered" with the TransferManasger, and allow transfers to specify the client to use by name (with overloads to make it easy to use a single default client and not have to worry or understand about the "registry" process or naming them).

Reply:
yes, repopulating the lookup map is something we need to brainstorm more in upcoming arch meetings.

I liked the design you showed in the last meeting, where we allow lookup (registration) using friendly names. I will add it as part of defining builder for the TransferMgr. 

The builder with configurations (e.g. block size, parallelism ..) + named client registration will be  a separate PR. As we talked in last sync, builder will be pushed after aggregation/download apis.